Sources/MongoMobile/MongoMobileLogger.swift

```diff
@@ -0,0 +1,99 @@
+/// If the user provided a logger when initializing the library, executes the 
+/// `onMessage` callback for that logger.
+internal func mongo_mobile_log_callback(userDataPtr: UnsafeMutableRawPointer?,
+                                        messagePtr: UnsafePointer<Int8>?,
+                                        componentPtr: UnsafePointer<Int8>?,
+                                        contextPtr: UnsafePointer<Int8>?,
+                                        severity: Int32) {
+    guard let logger = MongoMobile.logger else {
+        return
+    }
+
+    let message = String(from: messagePtr)
+    let component = String(from: componentPtr)
+    let context = String(from: contextPtr)
+    // swiftlint:disable:next force_unwrapping - server should always gives a valid raw value.
+    let severity = LogSeverity(rawValue: severity)!
```

Comment:
alternatively, we could be less restrictive on the values that go into `debug`, and just accept any Int32 there?

Reply:
That also sounds good. If we do add a catch-all mechanism (in either manner), we should also probably add an `assert` so that debug builds will start to fail if the server makes changes.

Reply:
I'm fine with either approach. Definitely agree that we should assert to catch server changes.